[libgeotiff] build failure on x64-linux-dynamic

**Host Environment**
- OS: Linux
- Compiler: g++

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install libgeotiff:x64-linux-dynamic --overlay-triplets=custom-triplets`

**Failure logs**
```
Installing 15/40 libgeotiff:x64-linux-dynamic...
Building libgeotiff[core]:x64-linux-dynamic...
-- [OVERLAY] Loading triplet configuration from: /home/l23394/Libs/vcpkg-2022.06.15/custom-triplets/x64-linux-dynamic.cmake
-- Downloading https://github.com/OSGeo/libgeotiff/archive/1.7.1.tar.gz -> OSGeo-libgeotiff-1.7.1.tar.gz...
-- Extracting source /home/l23394/Libs/vcpkg-2022.06.15/downloads/OSGeo-libgeotiff-1.7.1.tar.gz
-- Applying patch cmakelists.patch
-- Applying patch skip-doc-install.patch
-- Using source at /home/l23394/Libs/vcpkg-2022.06.15/buildtrees/libgeotiff/src/1.7.1-f5da4bf521.clean
-- Configuring x64-linux-dynamic-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:83 (message):
    Command failed: /snap/cmake/1088/bin/cmake /home/l23394/Libs/vcpkg-2022.06.15/buildtrees/libgeotiff/src/1.7.1-f5da4bf521.clean/libgeotiff -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/home/l23394/Libs/vcpkg-2022.06.15/packages/libgeotiff_x64-linux-dynamic/debug -DGEOTIFF_BIN_SUBDIR=bin -DWITH_TIFF=1 -DHAVE_TIFFOPEN=1 -DHAVE_TIFFMERGEFIELDINFO=1 -DCMAKE_MACOSX_BUNDLE=0 -DWITH_JPEG=OFF -DWITH_UTILITIES=OFF -DCMAKE_MAKE_PROGRAM=/home/l23394/Libs/vcpkg-2022.06.15/downloads/tools/ninja/1.10.2-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=ON -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/l23394/Libs/vcpkg-2022.06.15/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux-dynamic -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/l23394/Libs/vcpkg-2022.06.15/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/home/l23394/Libs/vcpkg-2022.06.15 -D_VCPKG_INSTALLED_DIR=/home/l23394/Libs/vcpkg-2022.06.15/installed -DVCPKG_MANIFEST_INSTALL=OFF
    Working Directory: /home/l23394/Libs/vcpkg-2022.06.15/buildtrees/libgeotiff/x64-linux-dynamic-dbg
    Error code: 1
    See logs for more information:
      /home/l23394/Libs/vcpkg-2022.06.15/buildtrees/libgeotiff/config-x64-linux-dynamic-dbg-out.log
      /home/l23394/Libs/vcpkg-2022.06.15/buildtrees/libgeotiff/config-x64-linux-dynamic-dbg-err.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:250 (vcpkg_execute_required_process)
  ports/libgeotiff/portfile.cmake:18 (vcpkg_cmake_configure)
  scripts/ports.cmake:146 (include)


error: building libgeotiff:x64-linux-dynamic failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libgeotiff
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[libgeotiff]+Build+error
Include '[libgeotiff] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-06-15-b0f08e40f8e8752dd0b260138ee078bd9599559c
    vcpkg-scripts version: unknown
```
You can also use the prefilled template from /home/l23394/Libs/vcpkg-2022.06.15/installed/vcpkg/issue_body.md.

----------------------------------------------------------------------------------------------------------------------------------------------------------------------
23394@23394:~/Libs/vcpkg-2022.06.15$ **cat /home/l23394/Libs/vcpkg-2022.06.15/buildtrees/libgeotiff/config-x64-linux-dynamic-dbg-err.log**
```
CMake Error at /home/l23394/Libs/vcpkg-2022.06.15/installed/x64-linux-dynamic/share/unofficial-sqlite3/unofficial-sqlite3-targets.cmake:55 (set_target_properties):
  The link interface of target "unofficial::sqlite3::sqlite3" contains:

    Threads::Threads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /home/l23394/Libs/vcpkg-2022.06.15/installed/x64-linux-dynamic/share/unofficial-sqlite3/unofficial-sqlite3-config.cmake:7 (include)
  /home/l23394/Libs/vcpkg-2022.06.15/scripts/buildsystems/vcpkg.cmake:824 (_find_package)
  /snap/cmake/1088/share/cmake-3.23/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /home/l23394/Libs/vcpkg-2022.06.15/installed/x64-linux-dynamic/share/proj/proj-config.cmake:13 (find_dependency)
  /home/l23394/Libs/vcpkg-2022.06.15/scripts/buildsystems/vcpkg.cmake:824 (_find_package)
  CMakeLists.txt:118 (FIND_PACKAGE)
```
**Additional context**
I used a copy of triplets/x64-linux.cmake named x64-linux-dynamic.cmake which I set the VCPKG_LIBRARY_LINKAGE to dynamic.

~~~
CMake Error at /home/l23394/Libs/vcpkg-2022.06.15/installed/x64-linux-dynamic/share/unofficial-sqlite3/unofficial-sqlite3-targets.cmake:55 (set_target_properties):
The link interface of target "unofficial::sqlite3::sqlite3" contains:

Threads::Threads
but the target was not found.
~~~

The actual issue in port sqlite3. It should link with `PRIVATE` here:
https://github.com/microsoft/vcpkg/blob/c168ce4e765409509c9c482faa0468c4e1ab386b/ports/sqlite3/CMakeLists.txt#L128-L131
or not limit this `find_dependency` to static linkage:
https://github.com/microsoft/vcpkg/blob/c168ce4e765409509c9c482faa0468c4e1ab386b/ports/sqlite3/sqlite3-config.in.cmake#L2-L5

--- a/ports/sqlite3/sqlite3-config.in.cmake
+++ b/ports/sqlite3/sqlite3-config.in.cmake
@@ -1,5 +1,5 @@
 
-if("@VCPKG_LIBRARY_LINKAGE@" STREQUAL "static" AND NOT WIN32)
+if(NOT WIN32)
     include(CMakeFindDependencyMacro)
     find_dependency(Threads)
 endif()

--- a/ports/sqlite3/vcpkg.json
+++ b/ports/sqlite3/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "sqlite3",
   "version": "3.37.2",
-  "port-version": 1,
+  "port-version": 2,
   "description": "SQLite is a software library that implements a self-contained, serverless, zero-configuration, transactional SQL database engine.",
   "homepage": "https://github.com/sqlite/sqlite",
   "license": "blessing",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6798,7 +6798,7 @@
     },
     "sqlite3": {
       "baseline": "3.37.2",
-      "port-version": 1
+      "port-version": 2
     },
     "sqlitecpp": {
       "baseline": "3.1.1",

--- a/versions/s-/sqlite3.json
+++ b/versions/s-/sqlite3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "8b96d4235bae7daf0cf3f65f66f9c28f9290628a",
+      "version": "3.37.2",
+      "port-version": 2
+    },
+    {
       "git-tree": "f49ad405225c6326558adb79f2082814002a8cda",
       "version": "3.37.2",
       "port-version": 1

